hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/StoreFlushContext.java

```diff
@@ -29,15 +29,25 @@
  * A store flush context carries the state required to prepare/flush/commit the store's cache.
  */
 @InterfaceAudience.Private
-interface StoreFlushContext {
+public abstract class StoreFlushContext {
+
+  protected HStore store;
+  protected long cacheFlushSeqNum;
+  protected FlushLifeCycleTracker tracker;
+
+  public StoreFlushContext(HStore store, Long cacheFlushSeqNum, FlushLifeCycleTracker tracker){
```

Comment:
It looks like it would take some work to do it, but can we make this be a `Store` instead of `HStore`?

It looks like most of the methods called by DefaultStoreFileContext are actually defined on HStore right now.

Reply:
Could do it. The idea was just to move what was already implemented as an HStore inner class to its own outer class, as we would already benefit from extending it for the specific persisted store flush context.